book/src/component-model/tasks.md

```diff
@@ -0,0 +1,36 @@
+## Tasks
+
+A `Task` records that an instance of a process was run in a Harbor environment. It may be helpful to
+think of a task as unit of work. The enrichment engine is implemented as a set of services that 
+implement the `TaskProvider` trait. You can build a service that implements the `TaskProvider` 
+trait and unless you override the key default functions, Harbor will record details about each 
+instantiation of the process by storing an instance of a `Task` entity.
+
+The `TaskProvider` trait has the following interface. 
+
+```rust
+pub trait TaskProvider: Service<Task> {
+    /// Implement this with your custom logic.
+    async fn run(&self, task: &mut Task) -> Result<HashMap<String, String>, Error>;
+
+    /// Runs the transaction script and store the results. Usually invoked by a CLI command handler.
+    async fn execute(&self, task: &mut Task) -> Result<(), Error> {
+        // See codebase for default implementation. You should probably not need to override this.
+    }
+
+    /// Inserts the [Task] record at the start of the transaction script.
+    async fn init(&self, task: &mut Task) -> Result<(), Error> {
```

Comment:
So, what does setting one of these tasks up to run look like?  I feel like it should just be like:
```let task = Task { ... }
let task_provider = TaskProvider { ... }
task_provider.execute(task)```

or is it that we need to run init() and complete() as well:
```let task = Task { ... }
let task_provider = TaskProvider { ... }
task_provider.init();
task_provider.execute(task)
task_provider.complete();```?

Just trying to hash out what this looks like in practice

Reply:
It is just as you suggest.

```rust
let task_provider = TaskProvider { ... }
task_provider.execute(task)```
```

`execute` handles invoking all the inner functions in the right order.  I'm adding an example callout so that's clear.